core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2981,12 +2978,9 @@ struct ViewValueFunctor<DeviceType, ValueType, false /* is_scalar */> {
       if (Kokkos::Profiling::profileLibraryLoaded()) {
         Kokkos::Profiling::endParallelFor(kpID);
       }
-    }
-#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
-    else {
+    } else {
```

Comment:
I mean that this branch should be deleted and the corresponding check above as well.